### PR TITLE
Fix signature check script

### PR DESCRIPTION
**Pull request by izchen**
When args contains `--spark`, `pv.register_presto_signatures()` is called incorrectly.

It is necessary to execute `pv.clear_signatures()` before `pv.register_spark_signatures()`, because [the presto functions are registered when `import pyvelox.pyvelox as pv` is executed](https://github.com/facebookincubator/velox/blob/fe5feccb3b6ed01e1c9d1bfd56e020f00a2b09e8/pyvelox/pyvelox.cpp#L238).

### CODE REVIEW

**Comment by izchen** [comment]
Fix this for https://github.com/facebookincubator/velox/pull/4922#issuecomment-1751525954

File: scripts/signature.py [hunk 1]
```diff
@@ -169,7 +171,7 @@ def parse_args(args):
     command = parser.add_subparsers(dest="command")
     export_command_parser = command.add_parser("export")
     export_command_parser.add_argument("--spark", action="store_true")
-    export_command_parser.add_argument("--presto", action="store_false")
+    export_command_parser.add_argument("--presto", action="store_true")
```
Comment (kgpai) [inline scripts/signature.py:174]: If I am not mistaken , store_false means presto defaults to true if no flag of spark presto is specified right ? 
Our default was to have presto enabled by default and thus the store_false, which is probably why register_presto_signatures gets called even with spark enabled. Thus I think just changing store_false to store_true aught to be enough. 

(Also If we would like both spark and presto signatures in one registry  , calling clear signatures would not work.)

**Comment by kgpai** [comment]
@izchen Can you try this out without the call to clear signatures ?

**Comment by izchen** [comment]
> @izchen Can you try this out without the call to clear signatures ?

```shell
$ python ./scripts/signature.py export --presto 1.json
$ python ./scripts/signature.py export 2.json
$ cat 1.json | cut -c -100
{"json_format": ["(json) -> varchar"], "repeat": ["(T,integer) -> array(T)"], "zip_with": ["(array(T
$ cat 2.json | cut -c -100
{"json_format": ["(json) -> varchar"], "repeat": ["(T,integer) -> array(T)"], "zip_with": ["(array(T
$ comm -3 1.json 2.json | wc -l
0
```

The presto signature is registered before `pv.register_presto_signatures()` is called. 

It is registered when `import pyvelox.pyvelox as pv` is called: https://github.com/facebookincubator/velox/blob/fe5feccb3b6ed01e1c9d1bfd56e020f00a2b09e8/pyvelox/pyvelox.cpp#L238

**Comment by izchen** [comment]
> Also If we would like both spark and presto signatures in one registry , calling clear signatures would not work.

I modified the script to support both spark and presto signatures in one registry

**Review by kgpai — APPROVED** [review]
Thanks for this fix @izchen !

**Pull request closed without merging**